calico_kubernetes/calico_kubernetes.py

```diff
@@ -326,16 +353,33 @@ def _apply_tags(self, profile_name, pod):
             print('No labels found in pod %s' % pod)
             return
 
+        try:
+            profile = self._datastore_client.get_profile(profile_name)
+        except KeyError:
+            print('Could not find profile with name %s' % profile_name)
+            sys.exit(1)
+
         for k, v in labels.iteritems():
             tag = '%s_%s' % (k, v)
             tag = tag.replace('/', '_')
             print('Adding tag ' + tag)
-            try:
-                self.calicoctl('profile', profile_name, 'tag', 'add', tag)
-            except sh.ErrorReturnCode as e:
-                print('Could not create tag %s.\n%s' % (tag, e))
+            profile.tags.add(tag)
+
+        self._datastore_client.profile_update_tags(profile)
+
         print('Finished applying tags.')
 
+    def _get_container_id(self, container_name):
+        try:
+            info = self._docker_client.inspect_container(container_name)
+        except docker.errors.APIError as e:
+            if e.response.status_code == 404:
+                print "Container %s was not found." % container_name
+            else:
+                print e.message
```

Comment:
"ERROR: Could not get container ID - %s" % e.message

Might be a more clear error.